lib/unpack.js

```diff
@@ -0,0 +1,59 @@
+module.exports = function unpack(obj, json) {
+    Object.keys(obj).forEach(prop => {
+        // ensure the json object has this property
+        if (!json.hasOwnProperty(prop)) {
+            // throw error if the json does not have an expected property
+        }
+        else {
+            // Need to start with array b/c arrays are also seen as objects
+            if (Array.isArray(obj[prop])) {
+                if (!Array.isArray(json[prop])) {
+                    // throw error if an array is expected but one is not provided.
+                }
+                else {   
+                    unpackArray(obj[prop], json[prop]);
+                }
+            }
+            // Check if something is an object
+            else if (obj[prop] === Object(obj[prop])) {
```

Comment:
curious where you found how to check if an object is an object like that, link? i coulndt find anything like that